packages/core/scripts/local/ClaimAllRewards.js

```diff
@@ -1,72 +1,190 @@
+// Usage: $(npm bin)/truffle exec ./scripts/ClaimAllRewards.js --batcherAddress 0x82458d1C812D7c930Bb3229c9e159cbabD9AA8Cb --network mainnet_mnemonic
+// Other optional args:
+// --round <round id> Example: `--round 9344 --round 9342`. This only considers rewards for rounds 9344 and 9342.
+// --claimAddress <address> Example: `--claimAddress 0x1234 --claimAddress 0x5678`. This only considers rewards for
+//   0x1234 and 0x5678. Defaults to claiming for _all_ addresses.
+// --excludeAddress <address> Example: `--excludeAddress 0x1234 --excludeAddress 0x5678`. This excludes addresses
+//   0x1234 and 0x5678. Defaults to excluding no addresses.
+// --onlyPrint. This flag takes no arguments. It tells the script to just print out the available rewards rather than
+//   trying to claim them. Default off.
+// --version <version string>. Example: `--version 1.2.2 --version latest` DVM versions to consider when claiming.
+//   Since a new Voting contract was deployed in version 2.0.0, that version will only claim for that contract, not
+//   older ones. Defaults to 1.2.2 and latest.
 
-const Voting = artifacts.require("Voting");
-const VotingInterfaceTesting = artifacts.require("VotingInterfaceTesting");
+const { getAbi, getAddress } = require("../../index");
 const TransactionBatcher = artifacts.require("TransactionBatcher");
+const lodash = require("lodash");
+const winston = require("winston");
+const { GasEstimator } = require("@uma/financial-templates-lib");
 
-const argv = require("minimist")(process.argv.slice(), { string: ["round", "batcherAddress"] });
+const argv = require("minimist")(process.argv.slice(), {
+  string: ["round", "batcherAddress", "claimAddress", "excludeAddress", "version"],
+  boolean: "onlyPrint"
+});
 
-const { toBN, toWei } = web3.utils;
+const { toBN, toWei, toChecksumAddress } = web3.utils;
 
 // This script claims all voter's rewards for the round provided.
 async function claimRewards() {
-  const voting = await VotingInterfaceTesting.at((await Voting.deployed()).address);
+  const rounds = argv.round ? lodash.castArray(argv.round) : [];
+  const excludeAddresses = argv.excludeAddress ? lodash.castArray(argv.excludeAddress) : [];
+  const claimAddresses = argv.claimAddress ? lodash.castArray(argv.claimAddress) : [];
+  const versions = argv.version ? lodash.castArray(argv.version) : ["latest", "1.2.2"];
 
-  const events = await voting.contract.getPastEvents("VoteRevealed", {
-    filter: { roundId: argv.round },
-    fromBlock: 0,
-    toBlock: "latest"
+  const account = (await web3.eth.getAccounts())[0];
+  const networkId = await web3.eth.net.getId();
+
+  const votingContracts = versions.map(version => {
+    if (version.startsWith("1")) {
+      return new web3.eth.Contract(getAbi("Voting", version), getAddress("Voting", networkId, version));
+    } else {
+      return new web3.eth.Contract(
+        getAbi("VotingAncillaryInterfaceTesting", "latest"),
```

Comment:
why do we use testing interface on v2? might want to leave a comment here. i cant even remember why and i remember doing something like this in voter dapp

Reply:
Will add a comment. TLDR: the newer voting contract has overloads (web3 picks blindly between the overloads) and we want to make sure to _only_ use the ancillary data methods throughout.